halo2_proofs/src/dev/cost_model.rs

```diff
@@ -0,0 +1,364 @@
+//! The cost estimator takes high-level parameters for a circuit design, and estimates the
+//! verification cost, as well as resulting proof size.
+
+use std::{cmp, iter, num::ParseIntError, str::FromStr};
+
+use crate::plonk::Circuit;
+use ff::{Field, FromUniformBytes};
+
+use super::MockProver;
+
+/// Supported commitment schemes
+#[derive(Debug)]
+pub enum CommitmentScheme {
+    /// Inner Product Argument commitment scheme
+    IPA,
+    /// KZG commitment scheme
+    KZG,
+}
+
+/// Options to build a circuit specifiction to measure the cost model of.
+#[derive(Debug)]
+pub struct CostOptions {
```

Comment:
We should include `::shuffle` arguments as well

Reply:
Do you have a pointer to docs of how the shuffle argument works? Looking at the code it seems that it only uses two rotations (`\omega X` and `\omega`), is that right? [67440c9](https://github.com/privacy-scaling-explorations/halo2/pull/235/commits/67440c9ca19d59a1e0943e391359a5c7b112f902)

Reply:
I think the feature is defined in the README (not sure if it points to any extra docs). IIRC the cost is to commit to the shuffled column which means extra commitment. 

Unsure about the rotations though. ping @kilic

Reply:
> er to docs of how the shuffle argument works? Looking at the code it seems that it only uses two rotations (`\omega X` and `\omega`), is that right? [67440c9](https://github.com/privacy-scaling-explorations/halo2/pull/235/commits/67440c9ca19d59a1e0943e391359a5c7b112f902)

@iquerejeta  Here it goes https://github.com/privacy-scaling-explorations/halo2/pull/185. It's kind of simplified version of lookup argument

Reply:
Great, thanks. And I think the computation of the lookup queries was wrong. Fixed it [here](https://github.com/privacy-scaling-explorations/halo2/pull/235/commits/c1184bb5286a3e07af945a2cd2c06691e52b0d0e#diff-a06d30f7a4e52c2456bb42ceaef21fabc01c3095725647e6695ebf828b50b0dcR69)